### PR TITLE
Add alembic

### PR DESCRIPTION
**Pull request by tomasmor42**
#77

### CODE REVIEW

File: alembic/env.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+fileConfig(config.config_file_name)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
+# target_metadata = None
+from api.models import db
```
Comment (MrNaif2018) [inline alembic/env.py]: Does that actually work? `alembic upgrade head` worked for me only with sys.path.append('.') code.

File: alembic/env.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
+
+    Calls to context.execute() here emit the given string to the
+    script output.
+
+    """
+    url = config.get_main_option("sqlalchemy.url")
```
Comment (MrNaif2018) [inline alembic/env.py]: I think it should be changed to `CONNECTION_STR` here too

File: alembic/env.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+def run_migrations_online():
+    """Run migrations in 'online' mode.
+
+    In this scenario we need to create an Engine
+    and associate a connection with the context.
+
+    """
+    #import pdb; pdb.set_trace()
```
Comment (MrNaif2018) [inline alembic/env.py]: Not neccesary, but remove this please (:

File: requirements/base.txt [hunk 4]
```diff
@@ -1,2 +1,3 @@
 # .env file reading
-python-decouple==3.1
\ No newline at end of file
+python-decouple==3.1
+alembic==1.2.1
```
Comment (MrNaif2018) [inline requirements/base.txt]: Good! But please add that to requirements/web.txt instead.
Also add sync driver `psycopg2-binary`

File: alembic/env.py (hunk 1 above)
Comment (tomasmor42) [reply alembic/env.py]: I set PYTHONPATH to a project directory, but it's a common thing to do, didn't know it's not the case here. 
(might now work with Docker, but I didn't have a chance to check :) ) 
Will try to switch to absolute paths.

Comment (MrNaif2018) [reply alembic/env.py]: Yes, PYTHONPATH works, but we shoudn't set it each time, better set it once in code.

File: alembic/env.py [hunk 5]
```diff
@@ -41,6 +39,9 @@ def run_migrations_offline():
     script output.
 
     """
+    alembic_config = config.get_section(config.config_ini_section)
+    alembic_config['sqlalchemy.url'] = CONNECTION_STR
+
     url = config.get_main_option("sqlalchemy.url")
```
Comment (MrNaif2018) [inline alembic/env.py]: Sorry I might not know about alembic a lot, but I think you just need to pass `CONNECTION_STR` to context.configure. Does this way work too? If yes, resolve that review comment.

File: alembic/env.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+fileConfig(config.config_file_name)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
+# target_metadata = None
+from main import db, CONNECTION_STR
```
Comment (MrNaif2018) [inline alembic/env.py]: ModuleNotFoundError: No module named 'main'
```suggestion
import sys
sys.path.insert(0, '.')
from api.db import db, CONNECTION_STR
```

File: alembic/env.py (hunk 5 above)
Comment (MrNaif2018) [reply alembic/env.py]: Seems fine, sorry

**Comment by MrNaif2018** [comment]
Good work! Merging! Would you like to be mentioned in contributors list I am going to create?
Which name do I write? Your github username?
Thank you! Looking forward to see you contributing to bitcart in future!

**Comment by tomasmor42** [comment]
Thanks a lot! Github username is fine. 
Good luck with the project! Will try to help in the future :)

**Comment by MrNaif2018** [comment]
Thank you (:

**Comment by MrNaif2018** [comment]
@all-contributors please add @MrNaif2018 for maintenance, code and doc.

**Comment by MrNaif2018** [comment]
@all-contributors please add @tomasmor42  for code.
P.S. Now you are in contributors list (:

**Pull request merged**